File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        i = 0
        j = 0
        n = len(prices) - 1
        ans = 0
        while j <=n:

            if prices[i] <= prices[j]:
                ans  = max(ans, prices[j]-prices[i])
                j = j  + 1
            else:
                while i < j and prices[i] > prices[j]:
                    i = i + 1
        return ans           

File: Data Structures & Algorithms/maximum-subarray/submission-0.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int mx = nums[0];
        int sm = 0;
        for(int i = 0; i < nums.size(); i++) {
            sm+=nums[i];
            if(sm <= 0) {
                mx = max(mx, nums[i]);
                sm = 0;
                
            }
            else mx = max(mx, sm);
            
        }
        return mx;
    }
};
